pykafka/pykafka/simpleconsumer.py

```diff
@@ -238,40 +247,49 @@ def fetch_offsets(self):
 
         reqs = [p.build_offset_fetch_request() for p in self._partitions.keys()]
         res = self._offset_manager.fetch_consumer_group_offsets(self._consumer_group, reqs)
-        parts_by_error = self._filter_partition_responses(res)
         error_handlers = {0: lambda op, pres: op.set_offset(pres.offset)}
-        self._handle_partition_errors(parts_by_error, error_handlers)
-        self._reset_offsets(
-            [a[0] for a in
-             parts_by_error[OffsetOutOfRangeError.ERROR_CODE]])
+        self._handle_partition_errors(res, handlers=error_handlers)
 
     def _filter_partition_responses(self, res):
         """Group partition responses by error code
 
-        This function accepts as input a FetchResponse or an OffsetFetchResponse.
+        This function accepts as input a Response instance
 
         :param res: a Response containing zero or more partition responses
-        :type res: FetchResponse or OffsetFetchResponse
+        :type res: pykafka.protocol.Response
         """
         partitions_by_error = defaultdict(list)
-        for partition_id, pres in res.topics[self._topic.name].iteritems():
-            owned_partition = self._partitions_by_id[partition_id]
-            partitions_by_error[pres.error].append((owned_partition, pres))
+        for topic_name in res.topics.keys():
+            for partition_id, pres in res.topics[topic_name].iteritems():
+                owned_partition = self._partitions_by_id[partition_id]
+                partitions_by_error[pres.error].append((owned_partition, pres))
         return partitions_by_error
 
-    def _handle_partition_errors(self, parts_by_error, handlers):
+    def _handle_partition_errors(self, res, handlers=None):
```

Comment:
That's a fine point, though knowledge of specific error codes is sprinkled all over the place.